src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseSpanBasedStringConcatTests.cs

```diff
@@ -0,0 +1,829 @@
+                yield return new object[]
+                {
+                    @"
+var _ = {|#0:Fwd({|#1:foo.Substring(1) + bar.Substring(1)|}) + Fwd({|#2:foo.Substring(1) + bar|}).Substring(1) + Fwd({|#3:foo + bar.Substring(1)|})|};",
+                    @"
+var _ = string.Concat(Fwd(string.Concat(foo.AsSpan(1), bar.AsSpan(1))), Fwd(string.Concat(foo.AsSpan(1), bar)).AsSpan(1), Fwd(string.Concat(foo, bar.AsSpan(1))));",
+                    new[] { 0, 1, 2, 3 },
+                    -4
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(Data_NestedViolations_CS))]
+        public Task NestedViolations_AreReportedAndFixed_CS(string testStatements, string fixedStatements, int[] locations, int? iterations = null)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = CSUsings + CSWithBody(testStatements),
+                FixedCode = CSUsings + CSWithBody(fixedStatements),
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                NumberOfIncrementalIterations = iterations,
+                NumberOfFixAllInDocumentIterations = iterations,
+                NumberOfFixAllIterations = iterations
+            };
+            test.ExpectedDiagnostics.AddRange(locations.Select(x => VerifyCS.Diagnostic(Rule).WithLocation(x)));
+            return test.RunAsync();
+        }
+
+        public static IEnumerable<object[]> Data_NestedViolations_VB
+        {
+            get
+            {
+                yield return new object[]
+                {
+                    @"
+Consume({|#0:Fwd({|#1:foo + bar.Substring(1)|}) + baz.Substring(1)|})",
+                    @"
+Consume(String.Concat(Fwd(String.Concat(foo, bar.AsSpan(1))), baz.AsSpan(1)))",
+                    new[] { 0, 1 }
+                };
+                yield return new object[]
+                {
+                    @"
+Dim s = {|#0:Fwd({|#1:foo.Substring(1) & bar.Substring(1)|}) & Fwd({|#2:foo.Substring(1) & bar|}).Substring(1) & Fwd({|#3:foo & bar.Substring(1)|})|}",
+                    @"
+Dim s = String.Concat(Fwd(String.Concat(foo.AsSpan(1), bar.AsSpan(1))), Fwd(String.Concat(foo.AsSpan(1), bar)).AsSpan(1), Fwd(String.Concat(foo, bar.AsSpan(1))))",
+                    new[] { 0, 1, 2, 3 }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(Data_NestedViolations_VB))]
+        public Task NestedViolations_AreReportedAndFixed_VB(string testStatements, string fixedStatements, int[] locations)
+        {
+            var test = new VerifyVB.Test
+            {
+                TestCode = VBUsings + VBWithBody(testStatements),
+                FixedCode = VBUsings + VBWithBody(fixedStatements),
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                NumberOfIncrementalIterations = -10,
+                NumberOfFixAllIterations = -10,
+                NumberOfFixAllInDocumentIterations = -10
```

Comment:
📝 If the default number of iterations isn't working, you should use _positive_ numbers for the replacement. Otherwise it's impossible to tell from reading the test exactly what the unexpected behavior will feel like to a user.

Reply:
Good to know. IIRC I think I set it to -10 because the order in which the violations were being fixed was inconsistent, which made it sometimes take different numbers of iterations. Is there any way to tell the test framework which order to apply the fixes in?

Reply:
Fixed.